File: antd/src/services/dnf-admin/accountsController.ts
// @ts-ignore
/* eslint-disable */
import { request } from 'umi';

/** 封禁 GET /api/accounts/disable/${param0} */
export async function disableAccounts(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.disableAccountsParams,
  options?: { [key: string]: any },
) {
  const { uid: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/accounts/disable/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 解封 GET /api/accounts/enable/${param0} */
export async function enableAccounts(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.enableAccountsParams,
  options?: { [key: string]: any },
) {
  const { uid: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/accounts/enable/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 账户详情 GET /api/accounts/info/${param0} */
export async function getAccounts(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getAccountsParams,
  options?: { [key: string]: any },
) {
  const { id: param0, ...queryParams } = params;
  return request<API.RAccounts>(`/api/accounts/info/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 开启全图王者 GET /api/accounts/openDungeon/${param0} */
export async function openDungeon(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.openDungeonParams,
  options?: { [key: string]: any },
) {
  const { uid: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/accounts/openDungeon/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 分页 POST /api/accounts/page */
export async function pageAccounts(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.pageAccountsParams,
  options?: { [key: string]: any },
) {
  return request<API.RListAccounts>('/api/accounts/page', {
    method: 'POST',
    params: {
      ...params,
      page: undefined,
      ...params['page'],
    },
    ...(options || {}),
  });
}

/** 点券充值 GET /api/accounts/rechargeBonds */
export async function rechargeBonds(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.rechargeBondsParams,
  options?: { [key: string]: any },
) {
  return request<API.RString>('/api/accounts/rechargeBonds', {
    method: 'GET',
    params: {
      // type has a default value: 1
      type: '1',

      // count has a default value: 1
      count: '1',
      ...params,
    },
    ...(options || {}),
  });
}

/** 重置角色创建限制 GET /api/accounts/resetCreateRole/${param0} */
export async function resetCreateRole(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.resetCreateRoleParams,
  options?: { [key: string]: any },
) {
  const { uid: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/accounts/resetCreateRole/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 重置密码 GET /api/accounts/resetPass/${param0} */
export async function resetPassword(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.resetPasswordParams,
  options?: { [key: string]: any },
) {
  const { uid: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/accounts/resetPass/${param0}`, {
    method: 'GET',
    params: {
      ...queryParams,
    },
    ...(options || {}),
  });
}

/** 添加账户 POST /api/accounts/save */
export async function saveAccounts(body: API.RegDto, options?: { [key: string]: any }) {
  return request<API.RLong>('/api/accounts/save', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 设置角色栏最大 GET /api/accounts/setMaxRole/${param0} */
export async function setMaxRole(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.setMaxRoleParams,
  options?: { [key: string]: any },
) {
  const { uid: param0, ...queryParams } = params;
  return request<API.RBoolean>(`/api/accounts/setMaxRole/${param0}`, {
    method: 'GET',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 编辑账户 POST /api/accounts/update */
export async function updateAccounts(body: API.Accounts, options?: { [key: string]: any }) {
  return request<API.RBoolean>('/api/accounts/update', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

File: be/src/main/java/plus/easydo/dnf/controller/AccountsController.java
package plus.easydo.dnf.controller;

import cn.dev33.satoken.annotation.SaCheckPermission;
import io.swagger.v3.oas.annotations.Operation;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import plus.easydo.dnf.dto.RegDto;
import plus.easydo.dnf.entity.Accounts;
import plus.easydo.dnf.qo.AccountsQo;
import plus.easydo.dnf.service.AccountsService;
import plus.easydo.dnf.vo.DataResult;
import plus.easydo.dnf.vo.R;

import java.util.List;


/**
 * 控制层。
 *
 * @author mybatis-flex-helper automatic generation
 * @since 1.0
 */
@RestController
@RequiredArgsConstructor
@RequestMapping("/api/accounts")
public class AccountsController {


    private final AccountsService accountsService;

    /**
     * 添加
     *
     * @param regDto
     * @return {@code true} 添加成功，{@code false} 添加失败
     */
    @Operation(summary = "添加账户")
    @SaCheckPermission("accounts.save")
    @PostMapping("/save")
    public R<Long> saveAccounts(@RequestBody RegDto regDto) {
        return DataResult.ok(accountsService.regAcc(regDto));
    }


    /**
     * 根据主键更新
     *
     * @param accounts
     * @return {@code true} 更新成功，{@code false} 更新失败
     */
    @Operation(summary = "编辑账户")
    @SaCheckPermission("accounts.update")
    @PostMapping("/update")
    public R<Boolean> updateAccounts(@RequestBody Accounts accounts) {
        return DataResult.ok(accountsService.update(accounts));
    }


    /**
     * 根据主键获取详细信息。
     *
     * @param id accounts主键
     * @return 详情
     */
    @Operation(summary = "账户详情")
    @SaCheckPermission("accounts")
    @GetMapping("/info/{id}")
    public  R<Accounts> getAccounts(@PathVariable Long id) {
        return DataResult.ok(accountsService.getById(id));
    }


    /**
     * 分页查询
     *
     * @param page 分页对象
     * @return 分页对象
     */
    @Operation(summary = "分页")
    @SaCheckPermission("accounts")
    @PostMapping("/page")
    public R<List<Accounts>> pageAccounts(AccountsQo page) {
        return DataResult.ok(accountsService.page(page));
    }

    /**
     * 解封账号
     *
     * @param uid uid
     * @return plus.easydo.dnf.vo.R<java.lang.String>
     * @author laoyu
     */
    @Operation(summary = "解封")
    @SaCheckPermission("accounts.enable")
    @GetMapping("/enable/{uid}")
    public R<Boolean> enableAccounts(
            @PathVariable(name = "uid") Long uid) {
        return DataResult.ok(accountsService.enableAcc(uid));
    }

    /**
     * 封禁账号
     *
     * @param uid uid
     * @return plus.easydo.dnf.vo.R<java.lang.String>
     * @author laoyu
     */
    @Operation(summary = "封禁")
    @SaCheckPermission("accounts.disable")
    @GetMapping("/disable/{uid}")
    public R<Boolean> disableAccounts(
            @PathVariable(name = "uid") Long uid) {
        return DataResult.ok(accountsService.disableAcc(uid));
    }

    /**
     * 重置密码
     *
     * @param uid uid
     * @param password password
     * @return plus.easydo.dnf.vo.R<java.lang.Boolean>
     * @author laoyu
     */
    @Operation(summary = "重置密码")
    @SaCheckPermission("accounts.resetPass")
    @GetMapping("/resetPass/{uid}")
    public R<Boolean> resetPassword(
            @PathVariable(name = "uid") Long uid,
            @RequestParam(name = "password") String password) {
        return DataResult.ok(accountsService.resetPassword(uid,password));
    }


    /**
     * 点券充值
     *
     * @param type type
     * @param uid uid
     * @param count count
     * @return plus.easydo.dnf.vo.R<java.lang.String>
     * @author laoyu
     * @date 2023-12-01
     */
    @Operation(summary = "点券充值")
    @SaCheckPermission("accounts.recharge")
    @GetMapping("/rechargeBonds")
    public R<String> rechargeBonds(
            @RequestParam(name = "type", defaultValue = "1") Integer type,
            @RequestParam(name = "uid") Long uid,
            @RequestParam(name = "count", defaultValue = "1") Long count ) {
        accountsService.rechargeBonds(type, uid, count);
        return DataResult.ok();
    }

    /**
     * 重置角色创建限制
     *
     * @param uid uid
     * @return plus.easydo.dnf.vo.R<java.lang.String>
     * @author laoyu
     * @date 2024/1/7
     */
    @Operation(summary = "重置角色创建限制")
    @SaCheckPermission("accounts.resetCreateRole")
    @GetMapping("/resetCreateRole/{uid}")
    public R<Boolean> resetCreateRole(@PathVariable(name = "uid") Long uid) {
        return DataResult.ok(accountsService.resetCreateRole(uid));
    }

    /**
     * 设置角色栏最大
     *
     * @param uid uid
     * @return plus.easydo.dnf.vo.R<java.lang.String>
     * @author laoyu
     * @date 2024/1/7
     */
    @Operation(summary = "设置角色栏最大")
    @SaCheckPermission("accounts.setMaxRole")
    @GetMapping("/setMaxRole/{uid}")
    public R<Boolean> setMaxRole(@PathVariable(name = "uid") Long uid) {
        return DataResult.ok(accountsService.setMaxRole(uid));
    }
    /**
     * 开启全图王者
     *
     * @param uid uid
     * @return plus.easydo.dnf.vo.R<java.lang.Boolean>
     * @author laoyu
     * @date 2024-01-31
     */
    @Operation(summary = "开启全图王者")
    @SaCheckPermission("accounts.openDungeon")
    @GetMapping("/openDungeon/{uid}")
    public R<Boolean> openDungeon(@PathVariable(name = "uid") Long uid) {
        return DataResult.ok(accountsService.openDungeon(uid));
    }

}
